extensions/aerosol_optics/mo_aerosol_optics.F90

```diff
@@ -0,0 +1,633 @@
+                                    tau_bcar_rh_table, ssa_bcar_rh_table, asy_bcar_rh_table, &
+                                    tau_bcar_table, ssa_bcar_table, asy_bcar_table, &
+                                    tau_ocar_rh_table, ssa_ocar_rh_table, asy_ocar_rh_table, &
+                                    tau_ocar_table, ssa_ocar_table, asy_ocar_table, &
+                                    tau, taussa, taussag)
+
+    integer,                               intent(in) :: ncol, nlay, nbnd, nbin, nrh
+    logical(wl), dimension(ncol,nlay),     intent(in) :: mask
+    integer,     dimension(ncol,nlay),     intent(in) :: type
+    real(wp),    dimension(ncol,nlay),     intent(in) :: size, mass, rh
+
+    real(wp),    dimension(nbin),          intent(in) :: aero_bin_lo, aero_bin_hi
+    real(wp),    dimension(nrh),           intent(in) :: aero_rh
+    integer,     dimension(2),             intent(in) :: rh_nsteps
+    real(wp),    dimension(2),             intent(in) :: rh_step_sizes, rh_offsets
+
+    real(wp),    dimension(nbin,    nbnd), intent(in) :: tau_dust_table, ssa_dust_table, asy_dust_table
+    real(wp),    dimension(nbin,nrh,nbnd), intent(in) :: tau_salt_table, ssa_salt_table, asy_salt_table
+    real(wp),    dimension(     nrh,nbnd), intent(in) :: tau_sulf_table, ssa_sulf_table, asy_sulf_table
+    real(wp),    dimension(     nrh,nbnd), intent(in) :: tau_bcar_rh_table, ssa_bcar_rh_table, asy_bcar_rh_table
+    real(wp),    dimension(         nbnd), intent(in) :: tau_bcar_table, ssa_bcar_table, asy_bcar_table
+    real(wp),    dimension(     nrh,nbnd), intent(in) :: tau_ocar_rh_table, ssa_ocar_rh_table, asy_ocar_rh_table
+    real(wp),    dimension(         nbnd), intent(in) :: tau_ocar_table, ssa_ocar_table, asy_ocar_table
+
+    real(wp),    dimension(ncol,nlay,nbnd)            :: tau, taussa, taussag
+    ! ---------------------------
+    integer  :: icol, ilay, ibnd, ibin
+    integer  :: index
+    real(wp) :: fint
+    real(wp) :: t, ts, tsg  ! tau, tau*ssa, tau*ssa*g
+    ! ---------------------------
+    !$acc parallel loop gang vector default(present) collapse(3)
+    !$omp target teams distribute parallel do simd collapse(3)
+    do ibnd = 1, nbnd
+      do ilay = 1,nlay
+        do icol = 1, ncol
+          ibin = get_aerosol_size_bin(nbin, aero_bin_lo, aero_bin_hi, size(icol,ilay))
+          if (mask(icol,ilay) .and. rh(icol,ilay) .le. rh_offsets(2)) then
+            index = min(floor((rh(icol,ilay) - rh_offsets(1))/rh_step_sizes(1))+1, rh_nsteps(1)-1)
+            fint = (rh(icol,ilay) - rh_offsets(1))/rh_step_sizes(1) - (index-1)
+          elseif (mask(icol,ilay) .and. rh(icol,ilay) .gt. rh_offsets(2)) then
+            index = min(floor((rh(icol,ilay) - rh_offsets(2))/rh_step_sizes(2))+1, rh_nsteps(2)-1)
+            fint = (rh(icol,ilay) - rh_offsets(2))/rh_step_sizes(2) - (index-1)
+          endif
+          ! dust
+          if (mask(icol,ilay) .and. type(icol,ilay) == merra_aero_dust) then
```

Comment:
@mjiacono SELECT CASE is a more transparent way to select from these 7 choices

Reply:
@RobertPincus Changed aerosol type selection to a SELECT(CASE) structure.